crates/sequencer/src/sequencer.rs

```diff
@@ -548,200 +555,91 @@ where
-                        match PostgresConnector::new(db_config).await {
-                            Ok(pg_connector) => {
-                                pg_connector
-                                    .insert_sequencer_commitment(
-                                        l1_start_height as u32,
-                                        l1_end_height as u32,
-                                        Into::<[u8; 32]>::into(tx_id).to_vec(),
-                                        commitment.l1_start_block_hash.to_vec(),
-                                        commitment.l1_end_block_hash.to_vec(),
-                                        l2_start,
-                                        l2_end,
-                                        commitment.merkle_root.to_vec(),
-                                        CommitmentStatus::Mempool,
-                                    )
-                                    .await
-                                    .expect("Sequencer: Failed to insert sequencer commitment");
-                            }
-                            Err(e) => {
-                                warn!("Failed to connect to postgres: {:?}", e);
-                            }
-                        }
+        let commitment_info = commitment_controller::get_commitment_info(
+            &self.ledger_db,
+            min_soft_confirmations_per_commitment,
+            prev_l1_height,
+        )?;
+
+        if let Some(commitment_info) = commitment_info {
+            debug!("Sequencer: enough soft confirmations to submit commitment");
+            let l2_range_to_submit = commitment_info.l2_height_range.clone();
+
+            // calculate exclusive range end
+            let range_end = BatchNumber(l2_range_to_submit.end().0 + 1); // cannnot add u64 to BatchNumber directly
+
+            let soft_confirmation_hashes = self
+                .ledger_db
+                .get_soft_batch_range(&(*l2_range_to_submit.start()..range_end))?
+                .iter()
+                .map(|sb| sb.hash)
+                .collect::<Vec<[u8; 32]>>();
+
+            let commitment = commitment_controller::get_commitment(
+                commitment_info.clone(),
+                soft_confirmation_hashes,
+            )?;
+
+            info!("Sequencer: submitting commitment: {:?}", commitment);
+
+            let blob = DaData::SequencerCommitment(commitment.clone())
+                .try_to_vec()
+                .map_err(|e| anyhow!(e))?;
+            let (notify, rx) = oneshot_channel();
+            let request = BlobWithNotifier { blob, notify };
+            inscription_queue
+                .send(request)
+                .map_err(|_| anyhow!("Bitcoin service already stopped!"))?;
+            let tx_id = rx
+                .await
+                .map_err(|_| anyhow!("DA service is dead!"))?
+                .map_err(|_| anyhow!("Send transaction cannot fail"))?;
```

Comment:
we should get da sercive stuff sorted out... getting unnecessarily complex

Reply:
https://github.com/chainwayxyz/citrea/issues/683